telegram/start_telegram_bot.js

```diff
@@ -480,17 +488,26 @@ module.exports = (args, cbk) => {
           return cbk();
         }
 
-        inquirer.prompt([interaction.user_id_prompt]).then(({code}) => {
-          if (!code) {
-            return cbk([400, 'ExpectedNumericConnectCode']);
-          }
-
-          connectedId = code;
-
-          return cbk();
-        });
+        return ask({
+          type: 'input',
+          message: interaction.user_id_prompt.message,
+          name: interaction.user_id_prompt.name,
+          validate: code => {
+            if (!code || !isNumber(code)) {
+              return `Expected Numeric Connect Code`;
+            }
+            const key = args.key;
 
-        return;
+            if (key.startsWith(code)) {
+              return `Expected Connect Code and not API Key or Bot Id`;
+            }
+              
+              connectedId = Number(code);
```

Comment:
I think this should be moved out of the validation function, it should still be set in the cbk of the ask